migrate-wordpress-directus/index.md

```diff
@@ -0,0 +1,244 @@
+```json
+{
+    "ID": "2",
+    "Title": "Sample Page",
+    "Content": "...",
+    "Excerpt": "...",
+    "Date": "2024-04-08 15:33:50",
+    "Post Type": "...",
+    "Permalink": "..."
+  },
+```
+### The URL replacer
+As I mentioned before, some images are in the posts and we need to extract those images and save them to Directus. After a file is saved to Directus, the JSON file is updated with the Directus image URL. Then, after the changes are completed, the new JSON file is saved locally.
+To start, first create a file with the name `extract-image.py`. The name is arbitrary.
+*Note: The Python file should be in the same directory as the JSON file.*
+We will start with the code below. Paste it in the file you just created.
+
+```python
+# Load JSON data from the file
+with open(json_file_path, "r") as file:
+    data = json.load(file)
+# Iterate through each post in the JSON data
+for post in data:
+    content = post["Content"]
+    # Check if there is an image in the content
+    if '<img src="' in content:
+        # Extract the image URL
+        start_index = content.index('<img src="') + 10
+        end_index = content.index('"', start_index)
+        image_url = content[start_index:end_index]
+        # Upload the image to Directus and get the new URL
+        new_image_url = upload_to_directus(image_url)
+        print(new_image_url)
+        # Replace the old image URL with the new one in the content
+        post["Content"] = content.replace(image_url, new_image_url)
+# Output the modified JSON data to a new file or print it out
+output_file_path = "modified_WP-data.json"
+with open(output_file_path, "w") as file:
+    json.dump(data, file, indent=4)
+# Print a message to indicate completion
+print("The JSON data has been updated with Directus image URLs.")
+```
+
+The code above opens and reads a JSON file specified by `json_file_path`. It then loops through `"Content"` field.  Within each post's content, the code checks for the presence of an `<img>` tag by looking for the substring `'<img src="'`. If such a substring is found, it means the post contains an image. The code then extracts the URL of this image by locating the substring that starts immediately after `'<img src="'` and ends at the next double-quote (`"`). 
+The extracted image URL is then passed to a function named `upload_to_directus` (this will be created next) to upload the image to Directus. The function returns `new_image_url` to this loop to replace the old URL in the post's content with this new URL.
+
+### The image uploader
+Now we will develop the function that will post the images to Directus and get the new URL. Paste the following code in above the code you have already.
+
+```python
+import json
+import requests
+import os
+# Replace with the path to your JSON file
+json_file_path = "WP-data.json" # Change this with the Pages JSON file to extract images for pages.
```

Comment:
I asked for this to be restructured slightly so it opened with the imports and link to the JSON file. https://github.com/directus-labs/guest-authoring/pull/127#discussion_r1562635642

Reply:
Yeah, I did that in "Creating Import Scripts" section. That was the section you marked previously.
I have put it in  "Replacing Image URLs" too since that section uses the file. This section is just a continuation of "Replacing Image URLs" and it doesn't use the file at all.

Reply:
Ohh yeah, corrected that